### PR TITLE
Convert media picker activity to kotlin

### PR DESCRIPTION
**Pull request by planarvoid**
This PR converts MediaPickerActivity to Kotlin in order to make future changes easier. There are no functionality changes

To test:
- Go to Media picker (add a new story)
- Check that it starts correctly
- Select items
- Make sure the items are selected correctly when you quit the activity

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/photopicker/mediapicker/MediaPickerActivity.kt [hunk 1]
```diff
@@ -0,0 +1,382 @@
...
+    }
+
+    override fun onMediaChosen(uriList: List<Uri>) {
+        if (uriList.size > 0) {
+            doMediaUrisSelected(uriList, APP_PICKER)
+        }
+    }
+
+    override fun onIconClicked(icon: MediaPickerIcon, allowMultipleSelection: Boolean) {
+        when (icon) {
```
Comment (develric) [inline WordPress/src/main/java/org/wordpress/android/ui/photopicker/mediapicker/MediaPickerActivity.kt:365]: maybe we could stay with an `if` here? really small np, so feel free to ignore 😊

**Review by develric — APPROVED** [review]
Hi @planarvoid , checked this one and it LGTM 👍 ! I just left a super minor np 😊 !

File: WordPress/src/main/java/org/wordpress/android/ui/photopicker/mediapicker/MediaPickerActivity.kt (hunk 1 above)
Comment (planarvoid) [reply WordPress/src/main/java/org/wordpress/android/ui/photopicker/mediapicker/MediaPickerActivity.kt:365]: my thinking was that if we add more actions, the `when` will let us know we missed it 🤷  but I'm fine with both

**Comment by planarvoid** [comment]
thanks for the review @develric ! let me know what you think about my answer. I'm cool with both approaches 👍

File: WordPress/src/main/java/org/wordpress/android/ui/photopicker/mediapicker/MediaPickerActivity.kt (hunk 1 above)
Comment (develric) [reply WordPress/src/main/java/org/wordpress/android/ui/photopicker/mediapicker/MediaPickerActivity.kt:365]: Thanks for sharing the reasoning behind. I'm fine with it 🙇‍♂️

**Comment by develric** [comment]
> thanks for the review @develric ! let me know what you think about my answer. I'm cool with both approaches 👍

Your reasoning makes sense to me, thanks for sharing 🙇‍♂️ ! :shipit:

**Pull request merged**